include/tvm/runtime/device_api.h

```diff
@@ -181,7 +181,38 @@ class TVM_DLL DeviceAPI {
    * \param ptr The pointer to be freed.
    */
   virtual void FreeWorkspace(TVMContext ctx, void* ptr);
-
+  /*!
+   * \brief Allocate a two dimensional texture data space on device
+   * \param ctx The device context to perform operation.
+   * \param width The width of the 2d texture in elements
+   * \param height The height of the 2d texture in elements
+   * \param type_hint The type of elements.
+   */
+  virtual void* AllocTexture(TVMContext ctx, size_t width, size_t height, DLDataType type_hint = {});
```

Comment:
Now that we have the Special scope API, likely we do not want to expose this as a separate DeviceAPI